nrf_security/doc/drivers.rst

```diff
@@ -0,0 +1,115 @@
+The Kconfig variable has a default value of 'y' when cc3xx is available in the SoC.
+
+.. _nrf_security_driver_oberon:
+
+nrf_oberon driver
+*****************
+
+The :ref:`nrf_oberon_readme` is distributed as a closed-source binary that provides select cryptographic algorithms optimized for use in nRF devices.
+This provides faster execution than the original Mbed TLS implementation.
+
+The nrf_oberon driver provides support for AES ciphers, SHA-1, SHA-256, SHA-384, SHA-512, ECDH and ECDSA using NIST curve secp224r1 and secp256r1, and ECJPAKE using NIST curve secp256r1.
+
+Enabling the nrf_oberon driver
+==============================
+
+To use the :ref:`nrf_oberon_readme` as a backend, set the :kconfig:`PSA_CRYPTO_DRIVER_OBERON` Kconfig variable to true.
+
+.. _nrf_security_backends_orig_mbedtls:
+
+Builtin Mbed TLS
+****************
+
+Builtin Mbed TLS uses the open-source software from the `Arm Mbed TLS project`_.
+
+The software is unaltered from the Arm Mbed TLS project source distribution, meaning it is in its `vanilla` form.
+
+You can use the builtin Mbed TLS support to add support for features not available in the `Arm CryptoCell cc3xx backend`_, for example AES-256 or ECC Brainpool curve types.
+
+Similarly, you can use the builtin Mbed TLS to add support for features not available in the `nrf_oberon backend`_, for example to support more ECC curve types.
+
+.. note:
+   Certain changes has been made tp the file ``pss_crypto_driver_wrappers`` to provide support for the PSA drivers.
+
+
+Enabling the builtin Mbed TLS support
+=====================================
+
+To enable the builtin Mbed TLS support, set the :kconfig:`PSA_CRYPTO_DRIVER_BUILTIN` Kconfig variable to true.
+
+It is generally not needed to enable builtin Mbed TLS support manually, as there is Kconfig logic that does that, depending on the enabled cryptographic features or algorithms and the configuration of PSA drivers.
+
+Legacy Mbed TLS
+***************
+
+Some legacy Mbed TLS APIs are still supported, for instance for TLS and DTLS support and backwards compatibility in Zephyr subsystems.
+
+Enabling legacy APIs requires enabling one of the available backends.
+
+.. note::
+   The legacy Mbed TLS APIs no longer support glued functionality and must choose one out of the selected backends
+   Legacy configurations has no effect on the configurations for the secure image of a TF-M build.
+
+Enabling legacy Mbed TLS support
+================================
+
+The legacy Mbed TLS APIs can be configured by setting the option kconfig:`NORDIC_SECURITY_BACKEND` instead of setting the option kconfig:`NRF_SECURITY`.
+
+Additionally the the options :kconfig:`CC3XX_BACKEND`or :kconfig:`OBERON_BACKEND` must be set, to enable the desired backend support.
+
+.. note::
+   Enabling the CryptoCell backend non-secure image of a TF-M build will have no effect.
```

Comment:
```suggestion
   Enabling the CryptoCell backend in a non-secure image of a TF-M build will have no effect.
```

Reply:
What about enabling cryptocell backend in a non-secure image for a non-TF-M build ?

Reply:
Changed to **Enabling the CryptoCell by using :kconfig:`CC3XX_BACKEND` in a non-secure image of a TF-M build will have no effect.** to make it absolutely clear that we are talking about a legacy (backend is only usable in scope of Legacy APIs and configuration)